### DISCUSSION TITLE
Need help with HTML +XPath for https://archive.is/searchURL

### DISCUSSION THREAD

**Discussion by JanWaldhorn** [FreshRSS/FreshRSS#6567] [Q&A] [answered]
I am trying to create an RSS feed of the articles of a certain website that are stored at [Archive.is ](https://archive.is/) using FressRss and HTML+XPath and unfortunately I am failing.

The error message is always:

> Damn! This feed has encountered a problem. Please make sure it is always readable and then update it.

I used FressRss version 1.24.1.
Does anyone have any tips? I have checked every single XPath with [xpather]( http://xpather.com/).
Below is the XML configuration.
```
<?xml version="1.0" encoding="UTF-8"?>
<opml xmlns:frss="https://freshrss.org/opml" version="2.0">
  <head>
    <title>FreshRSS | adminForge.de</title>
    <dateCreated>Sun, 16 Jun 2024 14:02:50 +0200</dateCreated>
  </head>
  <body>
    <outline text="Archive.is NYT" type="HTML+XPath" xmlUrl="https://archive.is/https://www.nytimes.com/books/first/b*" htmlUrl="https://archive.is/https://www.nytimes.com/books/first/b*" frss:xPathItem="//*[@id=&quot;CONTENT&quot;]//div[contains(@id, 'row')]" frss:xPathItemTitle="descendant::div[@class='THUMBS-BLOCK']//descendant::a/img/@title" frss:xPathItemContent="descendant::div[@class='THUMBS-BLOCK']//descendant::a/img/@title" frss:xPathItemUri="descendant::div[@class='THUMBS-BLOCK']//descendant::a/@href" frss:xPathItemAuthor="&quot;Unbekannt&quot;" frss:xPathItemTimestamp="descendant::div[@class='THUMBS-BLOCK']//descendant::a/div" frss:xPathItemThumbnail="descendant::div[@class='THUMBS-BLOCK']//descendant::a/img/@src" frss:xPathItemCategories="descendant::div[@class='THUMBS-BLOCK']//descendant::a/@href" frss:xPathItemUid="descendant::div[@class='THUMBS-BLOCK']//descendant::a/@href"/>
  </body>
</opml>

```
[f_64 Archive.is NYT.opml.txt](https://github.com/user-attachments/files/15857808/f_64.Archive.is.NYT.opml.txt)

**Comment by keynet** [comment]
I think you're having the same issue as me - it's a script created dynamic page and, as I understand it, curl is used to retrieve the page, - it doesn't know how to run a script, or doesn't trigger whatever is needed to generate the dynamic content.
Many websites use some kind of cache, so you don't see this, but this one doesn't.  Use CTRL+U to see page source.., different to looking at it once the browser has done its thing with scripts. 
The page I'm having trouble with is: https://haveyoursay.tfl.gov.uk/hub-page/healthy-streets. Mostly script and css, no content.  
Also appreciate any workarounds.

**Reply by Frenzie** [reply]
Maybe you can ask them?

**Reply by Alkarex** [reply]
@keynet Your case is completely different. The page requests the data through JSON, which returns the relevant information. For that, one can try FreshRSS' JSON Dot Notation mode.

However, your use-case has an API token, which probably expires, so a FreshRSS extension or another module such as in [RSS Bridge](https://github.com/RSS-Bridge/rss-bridge/) will likely be needed.

```sh
curl 'https://haveyoursay.tfl.gov.uk/api/v2/projects?page=1&per_page=30&filterable=true&fields%5Bproject%5D%5B%5D=name&fields%5Bproject%5D%5B%5D=image_url&fields%5Bproject%5D%5B%5D=banner_url&fields%5Bproject%5D%5B%5D=description&fields%5Bproject%5D%5B%5D=permalink&fields%5Bproject%5D%5B%5D=state&fields%5Bproject%5D%5B%5D=published_at&filters[sort][published_at]=desc&filters[state]=published&filters%5Btag_ids%5D%5B%5D=29093&filters%5Btag_ids%5D%5B%5D=30557&filters%5Btag_ids%5D%5B%5D=42234&filters%5Btag_ids%5D%5B%5D=42235&filters%5Btag_ids%5D%5B%5D=42236&filters%5Btag_ids%5D%5B%5D=30562&filters%5Btag_ids%5D%5B%5D=30563&filters%5Btag_ids%5D%5B%5D=30564&filters%5Btag_ids%5D%5B%5D=42237&filters%5Btag_ids%5D%5B%5D=42238&filters%5Btag_ids%5D%5B%5D=42239&filters%5Btag_ids%5D%5B%5D=42240&filters%5Btag_ids%5D%5B%5D=30566&filters%5Btag_ids%5D%5B%5D=41496&filters%5Btag_ids%5D%5B%5D=41497&filters%5Btag_ids%5D%5B%5D=42240&filters%5Btag_ids%5D%5B%5D=41932&filters[visibility_mode]=Public&filters[parent_id]=11342' \
--globoff -H 'x-api-token: Bearer eyJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MjE2Mzc1NDksImp0aSI6ImVmYmZiODRkMjNhM2E2ZWExYjA1YzUwYjUxNjhlY2YxIiwiZXhwIjoxNzIxODEwMzQ5LCJpc3MiOiJCYW5nIFRoZSBUYWJsZSBQdnQgTHRkIiwiZGF0YSI6eyJ1c2VyX2lkIjozMDQzMTMyMTEsInVzZXJfdHlwZSI6IkFub255bW91c1VzZXIiLCJzaXRlX2lkIjpudWxsLCJwcm9qZWN0X2lkIjpudWxsfX0.rbC1Df9RyHppCNnMqLhqT3F2LCpuBxHS1sWQa7itKrM'
```

**Reply by keynet** [reply]
Thanks - will give it a try

**Reply by keynet** [reply]
Just wondering how you came up with that API command line? It produces useful data. The web page has a large amount of script etc, makes it hard to identify the data in amongst it all.  Can you point me to any docs or examples of FreshRSS' JSON Dot Notation mode? Thx

**Reply by Alkarex** [reply]
See https://freshrss.github.io/FreshRSS/en/users/11_website_scraping.html
To find that call, have a look at your browser network view in the developer tools, and filter by JSON type.
One problem in FreshRSS at the moment, and preventing this use-case from working, is that we cannot pass arbitrary HTTP headers such as `x-api-token` but that is easy to add (PR welcome).
By the way, you can Base64-decode that x-api-token Bearer to inspect the JSON Token.

**Comment by Alkarex** [comment]
@JanWaldhorn Sorry for the delay. The Web page does contain the relevant information in HTML, but it is getting blocked by a captcha:

```
curl -v 'https://archive.is/https://www.nytimes.com/books/first/b*'
```

You may be able to solve it by changing the user agent and cookies in the feed options of FreshRSS, but that requires a bit of experimentation.

**Reply by JanWaldhorn** [reply, marked as answer]
Hi @Alkarex thanks for the reply and help.

I have solved my problem by changing the agent to
`Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0`

Would it perhaps be possible to display such an error (captcha) when creating the feed? Or at least point out the possibility (test URL with curl- v) ? I had tried and searched a lot before the message above, but didn't come up with the idea of `curl -v “url”`. (I know actually obvious....)

Anyway, I am very happy that it works now. Thank you very much!

**Reply by Alkarex** [reply]
We have a work-in-progress here https://github.com/FreshRSS/FreshRSS/discussions/6287 in order to produce a documentation to help people debug feed problems. Help welcome :-)
